2.3.4 build failure with Xcode 9.3 on macOS 10.13

Boost  1.67.0
https://gist.github.com/commitay/353a36ac506ffa735936a01c2d747155
```
[ 80%] Building CXX object CMakeFiles/osm2pgrouting.dir/src/parser/OSMDocumentParserCallback.cpp.o
/usr/local/Homebrew/Library/Homebrew/shims/super/clang++   -I/usr/local/opt/libpq/include -I/tmp/osm2pgrouting-20180419-60539-y3rfro/osm2pgrouting-2.3.4/src -I/usr/local/include -I/usr/local/opt/expat/include -I/tmp/osm2pgrouting-20180419-60539-y3rfro/osm2pgrouting-2.3.4/include  -std=c++14 -D_FILE_OFFSET_BITS=64 -DNDEBUG   -o CMakeFiles/osm2pgrouting.dir/src/parser/OSMDocumentParserCallback.cpp.o -c /tmp/osm2pgrouting-20180419-60539-y3rfro/osm2pgrouting-2.3.4/src/parser/OSMDocumentParserCallback.cpp
[ 84%] Building CXX object CMakeFiles/osm2pgrouting.dir/src/parser/XMLParser.cpp.o
/usr/local/Homebrew/Library/Homebrew/shims/super/clang++   -I/usr/local/opt/libpq/include -I/tmp/osm2pgrouting-20180419-60539-y3rfro/osm2pgrouting-2.3.4/src -I/usr/local/include -I/usr/local/opt/expat/include -I/tmp/osm2pgrouting-20180419-60539-y3rfro/osm2pgrouting-2.3.4/include  -std=c++14 -D_FILE_OFFSET_BITS=64 -DNDEBUG   -o CMakeFiles/osm2pgrouting.dir/src/parser/XMLParser.cpp.o -c /tmp/osm2pgrouting-20180419-60539-y3rfro/osm2pgrouting-2.3.4/src/parser/XMLParser.cpp
[ 88%] Building CXX object CMakeFiles/osm2pgrouting.dir/src/utilities/handle_pgpass.cpp.o
/usr/local/Homebrew/Library/Homebrew/shims/super/clang++   -I/usr/local/opt/libpq/include -I/tmp/osm2pgrouting-20180419-60539-y3rfro/osm2pgrouting-2.3.4/src -I/usr/local/include -I/usr/local/opt/expat/include -I/tmp/osm2pgrouting-20180419-60539-y3rfro/osm2pgrouting-2.3.4/include  -std=c++14 -D_FILE_OFFSET_BITS=64 -DNDEBUG   -o CMakeFiles/osm2pgrouting.dir/src/utilities/handle_pgpass.cpp.o -c /tmp/osm2pgrouting-20180419-60539-y3rfro/osm2pgrouting-2.3.4/src/utilities/handle_pgpass.cpp
In file included from /tmp/osm2pgrouting-20180419-60539-y3rfro/osm2pgrouting-2.3.4/src/parser/ConfigurationParserCallback.cpp:23:
In file included from /usr/local/include/boost/lexical_cast.hpp:30:
In file included from /usr/local/include/boost/range/iterator_range_core.hpp:38:
In file included from /usr/local/include/boost/range/functions.hpp:20:
In file included from /usr/local/include/boost/range/size.hpp:21:
In file included from /usr/local/include/boost/range/size_type.hpp:20:
In file included from /usr/local/include/boost/range/concepts.hpp:20:
In file included from /usr/local/include/boost/iterator/iterator_concepts.hpp:26:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/algorithm:643:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2051:9: error: cannot initialize a member subobject of type 'std::__1::basic_string<char> *' with an rvalue of type 'long'
      : __value_(_VSTD::forward<_Up>(__u)){};
        ^        ~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2151:9: note: in instantiation of function template specialization 'std::__1::__compressed_pair_elem<std::__1::basic_string<char> *, 0, false>::__compressed_pair_elem<long, void>' requested here
      : _Base1(std::forward<_U1>(__t1)), _Base2(std::forward<_U2>(__t2)) {}
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/vector:432:7: note: in instantiation of function template specialization 'std::__1::__compressed_pair<std::__1::basic_string<char> *, std::__1::allocator<std::__1::basic_string<char> > >::__compressed_pair<long, const std::__1::allocator<std::__1::basic_string<char> > &>' requested here
      __end_cap_(nullptr, __a)
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/vector:1193:7: note: in instantiation of member function 'std::__1::__vector_base<std::__1::basic_string<char>, std::__1::allocator<std::__1::basic_string<char> > >::__vector_base' requested here
    : __base(__alloc_traits::select_on_container_copy_construction(__x.__alloc()))
      ^
/usr/local/include/boost/program_options/errors.hpp:283:13: note: in instantiation of member function 'std::__1::vector<std::__1::basic_string<char>, std::__1::allocator<std::__1::basic_string<char> > >::vector' requested here
            m_alternatives(xalternatives)
            ^
In file included from /tmp/osm2pgrouting-20180419-60539-y3rfro/osm2pgrouting-2.3.4/src/parser/ConfigurationParserCallback.cpp:23:
In file included from /usr/local/include/boost/lexical_cast.hpp:30:
In file included from /usr/local/include/boost/range/iterator_range_core.hpp:38:
In file included from /usr/local/include/boost/range/functions.hpp:20:
In file included from /usr/local/include/boost/range/size.hpp:21:
In file included from /usr/local/include/boost/range/size_type.hpp:20:
In file included from /usr/local/include/boost/range/concepts.hpp:20:
In file included from /usr/local/include/boost/iterator/iterator_concepts.hpp:26:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/algorithm:643:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2051:9: error: cannot initialize a member subobject of type 'osm2pgr::Node **' with an rvalue of type 'long'
      : __value_(_VSTD::forward<_Up>(__u)){};
        ^        ~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2151:9: note: in instantiation of function template specialization 'std::__1::__compressed_pair_elem<osm2pgr::Node **, 0, false>::__compressed_pair_elem<long, void>' requested here
      : _Base1(std::forward<_U1>(__t1)), _Base2(std::forward<_U2>(__t2)) {}
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/vector:432:7: note: in instantiation of function template specialization 'std::__1::__compressed_pair<osm2pgr::Node **, std::__1::allocator<osm2pgr::Node *> >::__compressed_pair<long, const std::__1::allocator<osm2pgr::Node *> &>' requested here
      __end_cap_(nullptr, __a)
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/vector:1193:7: note: in instantiation of member function 'std::__1::__vector_base<osm2pgr::Node *, std::__1::allocator<osm2pgr::Node *> >::__vector_base' requested here
    : __base(__alloc_traits::select_on_container_copy_construction(__x.__alloc()))
      ^
/tmp/osm2pgrouting-20180419-60539-y3rfro/osm2pgrouting-2.3.4/include/osm_elements/Way.h:68:56: note: in instantiation of member function 'std::__1::vector<osm2pgr::Node *, std::__1::allocator<osm2pgr::Node *> >::vector' requested here
     const std::vector<Node*> nodeRefs() const {return m_NodeRefs;}
                                                       ^
In file included from /tmp/osm2pgrouting-20180419-60539-y3rfro/osm2pgrouting-2.3.4/src/parser/ConfigurationParserCallback.cpp:23:
In file included from /usr/local/include/boost/lexical_cast.hpp:30:
In file included from /usr/local/include/boost/range/iterator_range_core.hpp:38:
In file included from /usr/local/include/boost/range/functions.hpp:20:
In file included from /usr/local/include/boost/range/size.hpp:21:
In file included from /usr/local/include/boost/range/size_type.hpp:20:
In file included from /usr/local/include/boost/range/concepts.hpp:20:
In file included from /usr/local/include/boost/iterator/iterator_concepts.hpp:26:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/algorithm:643:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2051:9: error: cannot initialize a member subobject of type 'long long *' with an rvalue of type 'long'
      : __value_(_VSTD::forward<_Up>(__u)){};
        ^        ~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2151:9: note: in instantiation of function template specialization 'std::__1::__compressed_pair_elem<long long *, 0, false>::__compressed_pair_elem<long, void>' requested here
      : _Base1(std::forward<_U1>(__t1)), _Base2(std::forward<_U2>(__t2)) {}
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/vector:432:7: note: in instantiation of function template specialization 'std::__1::__compressed_pair<long long *, std::__1::allocator<long long> >::__compressed_pair<long, const std::__1::allocator<long long> &>' requested here
      __end_cap_(nullptr, __a)
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/vector:1193:7: note: in instantiation of member function 'std::__1::__vector_base<long long, std::__1::allocator<long long> >::__vector_base' requested here
    : __base(__alloc_traits::select_on_container_copy_construction(__x.__alloc()))
      ^
/tmp/osm2pgrouting-20180419-60539-y3rfro/osm2pgrouting-2.3.4/include/osm_elements/Relation.h:63:52: note: in instantiation of member function 'std::__1::vector<long long, std::__1::allocator<long long> >::vector' requested here
     std::vector<int64_t> way_refs() const {return m_WayRefs;}
                                                   ^
In file included from /tmp/osm2pgrouting-20180419-60539-y3rfro/osm2pgrouting-2.3.4/src/parser/ConfigurationParserCallback.cpp:23:
In file included from /usr/local/include/boost/lexical_cast.hpp:30:
In file included from /usr/local/include/boost/range/iterator_range_core.hpp:38:
In file included from /usr/local/include/boost/range/functions.hpp:20:
In file included from /usr/local/include/boost/range/size.hpp:21:
In file included from /usr/local/include/boost/range/size_type.hpp:20:
In file included from /usr/local/include/boost/range/concepts.hpp:20:
In file included from /usr/local/include/boost/iterator/iterator_concepts.hpp:26:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/algorithm:643:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2151:9: error: no matching constructor for initialization of '__compressed_pair_elem<std::__1::basic_string<char> *, 0>'
      : _Base1(std::forward<_U1>(__t1)), _Base2(std::forward<_U2>(__t2)) {}
        ^      ~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__split_buffer:309:7: note: in instantiation of function template specialization 'std::__1::__compressed_pair<std::__1::basic_string<char> *, std::__1::allocator<std::__1::basic_string<char> > &>::__compressed_pair<long, std::__1::allocator<std::__1::basic_string<char> > &>' requested here
    : __end_cap_(nullptr, __a)
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/vector:1567:49: note: in instantiation of member function 'std::__1::__split_buffer<std::__1::basic_string<char>, std::__1::allocator<std::__1::basic_string<char> > &>::__split_buffer' requested here
    __split_buffer<value_type, allocator_type&> __v(__recommend(size() + 1), size(), __a);
                                                ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/vector:1588:9: note: in instantiation of function template specialization 'std::__1::vector<std::__1::basic_string<char>, std::__1::allocator<std::__1::basic_string<char> > >::__push_back_slow_path<const std::__1::basic_string<char> &>' requested here
        __push_back_slow_path(__x);
        ^
/tmp/osm2pgrouting-20180419-60539-y3rfro/osm2pgrouting-2.3.4/include/database/table_management.h:83:16: note: in instantiation of member function 'std::__1::vector<std::__1::basic_string<char>, std::__1::allocator<std::__1::basic_string<char> > >::push_back' requested here
         m_sql.push_back(sql);
               ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2037:8: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'long' to 'const std::__1::__compressed_pair_elem<std::__1::basic_string<char> *, 0, false>' for 1st argument
struct __compressed_pair_elem {
       ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2037:8: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'long' to 'std::__1::__compressed_pair_elem<std::__1::basic_string<char> *, 0, false>' for 1st argument
struct __compressed_pair_elem {
       ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2050:3: note: candidate template ignored: substitution failure [with _Up = long, $1 = void]
  __compressed_pair_elem(_Up&& __u)
  ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2055:3: note: candidate constructor template not viable: requires 3 arguments, but 1 was provided
  __compressed_pair_elem(piecewise_construct_t, tuple<_Args...> __args,
  ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2043:39: note: candidate constructor not viable: requires 0 arguments, but 1 was provided
  _LIBCPP_INLINE_VISIBILITY constexpr __compressed_pair_elem() : __value_() {}
                                      ^
4 errors generated.
make[2]: *** [CMakeFiles/osm2pgrouting.dir/src/parser/ConfigurationParserCallback.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /tmp/osm2pgrouting-20180419-60539-y3rfro/osm2pgrouting-2.3.4/src/parser/OSMDocumentParserCallback.cpp:26:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string:470:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string_view:169:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__string:56:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/algorithm:643:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2051:9: error: cannot initialize a member subobject of type 'std::__1::basic_string<char> *' with an rvalue of type 'long'
      : __value_(_VSTD::forward<_Up>(__u)){};
        ^        ~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2151:9: note: in instantiation of function template specialization 'std::__1::__compressed_pair_elem<std::__1::basic_string<char> *, 0, false>::__compressed_pair_elem<long, void>' requested here
      : _Base1(std::forward<_U1>(__t1)), _Base2(std::forward<_U2>(__t2)) {}
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/vector:432:7: note: in instantiation of function template specialization 'std::__1::__compressed_pair<std::__1::basic_string<char> *, std::__1::allocator<std::__1::basic_string<char> > >::__compressed_pair<long, const std::__1::allocator<std::__1::basic_string<char> > &>' requested here
      __end_cap_(nullptr, __a)
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/vector:1193:7: note: in instantiation of member function 'std::__1::__vector_base<std::__1::basic_string<char>, std::__1::allocator<std::__1::basic_string<char> > >::__vector_base' requested here
    : __base(__alloc_traits::select_on_container_copy_construction(__x.__alloc()))
      ^
/usr/local/include/boost/program_options/errors.hpp:283:13: note: in instantiation of member function 'std::__1::vector<std::__1::basic_string<char>, std::__1::allocator<std::__1::basic_string<char> > >::vector' requested here
            m_alternatives(xalternatives)
            ^
In file included from /tmp/osm2pgrouting-20180419-60539-y3rfro/osm2pgrouting-2.3.4/src/parser/OSMDocumentParserCallback.cpp:26:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string:470:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string_view:169:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__string:56:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/algorithm:643:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2051:9: error: cannot initialize a member subobject of type 'osm2pgr::Node **' with an rvalue of type 'long'
      : __value_(_VSTD::forward<_Up>(__u)){};
        ^        ~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2151:9: note: in instantiation of function template specialization 'std::__1::__compressed_pair_elem<osm2pgr::Node **, 0, false>::__compressed_pair_elem<long, void>' requested here
      : _Base1(std::forward<_U1>(__t1)), _Base2(std::forward<_U2>(__t2)) {}
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/vector:432:7: note: in instantiation of function template specialization 'std::__1::__compressed_pair<osm2pgr::Node **, std::__1::allocator<osm2pgr::Node *> >::__compressed_pair<long, const std::__1::allocator<osm2pgr::Node *> &>' requested here
      __end_cap_(nullptr, __a)
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/vector:1193:7: note: in instantiation of member function 'std::__1::__vector_base<osm2pgr::Node *, std::__1::allocator<osm2pgr::Node *> >::__vector_base' requested here
    : __base(__alloc_traits::select_on_container_copy_construction(__x.__alloc()))
      ^
/tmp/osm2pgrouting-20180419-60539-y3rfro/osm2pgrouting-2.3.4/include/osm_elements/Way.h:68:56: note: in instantiation of member function 'std::__1::vector<osm2pgr::Node *, std::__1::allocator<osm2pgr::Node *> >::vector' requested here
     const std::vector<Node*> nodeRefs() const {return m_NodeRefs;}
                                                       ^
In file included from /tmp/osm2pgrouting-20180419-60539-y3rfro/osm2pgrouting-2.3.4/src/parser/OSMDocumentParserCallback.cpp:26:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string:470:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string_view:169:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__string:56:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/algorithm:643:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2051:9: error: cannot initialize a member subobject of type 'long long *' with an rvalue of type 'long'
      : __value_(_VSTD::forward<_Up>(__u)){};
        ^        ~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2151:9: note: in instantiation of function template specialization 'std::__1::__compressed_pair_elem<long long *, 0, false>::__compressed_pair_elem<long, void>' requested here
      : _Base1(std::forward<_U1>(__t1)), _Base2(std::forward<_U2>(__t2)) {}
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/vector:432:7: note: in instantiation of function template specialization 'std::__1::__compressed_pair<long long *, std::__1::allocator<long long> >::__compressed_pair<long, const std::__1::allocator<long long> &>' requested here
      __end_cap_(nullptr, __a)
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/vector:1193:7: note: in instantiation of member function 'std::__1::__vector_base<long long, std::__1::allocator<long long> >::__vector_base' requested here
    : __base(__alloc_traits::select_on_container_copy_construction(__x.__alloc()))
      ^
/tmp/osm2pgrouting-20180419-60539-y3rfro/osm2pgrouting-2.3.4/include/osm_elements/Relation.h:63:52: note: in instantiation of member function 'std::__1::vector<long long, std::__1::allocator<long long> >::vector' requested here
     std::vector<int64_t> way_refs() const {return m_WayRefs;}
                                                   ^
In file included from /tmp/osm2pgrouting-20180419-60539-y3rfro/osm2pgrouting-2.3.4/src/parser/OSMDocumentParserCallback.cpp:26:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string:470:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string_view:169:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__string:56:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/algorithm:643:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2151:9: error: no matching constructor for initialization of '__compressed_pair_elem<std::__1::basic_string<char> *, 0>'
      : _Base1(std::forward<_U1>(__t1)), _Base2(std::forward<_U2>(__t2)) {}
        ^      ~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__split_buffer:309:7: note: in instantiation of function template specialization 'std::__1::__compressed_pair<std::__1::basic_string<char> *, std::__1::allocator<std::__1::basic_string<char> > &>::__compressed_pair<long, std::__1::allocator<std::__1::basic_string<char> > &>' requested here
    : __end_cap_(nullptr, __a)
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/vector:1567:49: note: in instantiation of member function 'std::__1::__split_buffer<std::__1::basic_string<char>, std::__1::allocator<std::__1::basic_string<char> > &>::__split_buffer' requested here
    __split_buffer<value_type, allocator_type&> __v(__recommend(size() + 1), size(), __a);
                                                ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/vector:1588:9: note: in instantiation of function template specialization 'std::__1::vector<std::__1::basic_string<char>, std::__1::allocator<std::__1::basic_string<char> > >::__push_back_slow_path<const std::__1::basic_string<char> &>' requested here
        __push_back_slow_path(__x);
        ^
/tmp/osm2pgrouting-20180419-60539-y3rfro/osm2pgrouting-2.3.4/include/database/table_management.h:83:16: note: in instantiation of member function 'std::__1::vector<std::__1::basic_string<char>, std::__1::allocator<std::__1::basic_string<char> > >::push_back' requested here
         m_sql.push_back(sql);
               ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2037:8: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'long' to 'const std::__1::__compressed_pair_elem<std::__1::basic_string<char> *, 0, false>' for 1st argument
struct __compressed_pair_elem {
       ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2037:8: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'long' to 'std::__1::__compressed_pair_elem<std::__1::basic_string<char> *, 0, false>' for 1st argument
struct __compressed_pair_elem {
       ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2050:3: note: candidate template ignored: substitution failure [with _Up = long, $1 = void]
  __compressed_pair_elem(_Up&& __u)
  ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2055:3: note: candidate constructor template not viable: requires 3 arguments, but 1 was provided
  __compressed_pair_elem(piecewise_construct_t, tuple<_Args...> __args,
  ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:2043:39: note: candidate constructor not viable: requires 0 arguments, but 1 was provided
  _LIBCPP_INLINE_VISIBILITY constexpr __compressed_pair_elem() : __value_() {}
                                      ^
4 errors generated.
make[2]: *** [CMakeFiles/osm2pgrouting.dir/src/parser/OSMDocumentParserCallback.cpp.o] Error 1
make[1]: *** [CMakeFiles/osm2pgrouting.dir/all] Error 2
make: *** [all] Error 2
```

@cvvergara I can test with 9.3 in a bit, but testing with Xcode 8.2.1
```
bash-3.2$ make -j1
/usr/local/Cellar/cmake/3.11.1/bin/cmake -H/tmp/osm2pgrouting-20180419-86054-4qcrrl -B/tmp/osm2pgrouting-20180419-86054-4qcrrl/build --check-build-system CMakeFiles/Makefile.cmake 0
/usr/local/Cellar/cmake/3.11.1/bin/cmake -E cmake_progress_start /tmp/osm2pgrouting-20180419-86054-4qcrrl/build/CMakeFiles /tmp/osm2pgrouting-20180419-86054-4qcrrl/build/CMakeFiles/progress.marks
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f CMakeFiles/Makefile2 all
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f CMakeFiles/osm2pgrouting.dir/build.make CMakeFiles/osm2pgrouting.dir/depend
cd /tmp/osm2pgrouting-20180419-86054-4qcrrl/build && /usr/local/Cellar/cmake/3.11.1/bin/cmake -E cmake_depends "Unix Makefiles" /tmp/osm2pgrouting-20180419-86054-4qcrrl /tmp/osm2pgrouting-20180419-86054-4qcrrl /tmp/osm2pgrouting-20180419-86054-4qcrrl/build /tmp/osm2pgrouting-20180419-86054-4qcrrl/build /tmp/osm2pgrouting-20180419-86054-4qcrrl/build/CMakeFiles/osm2pgrouting.dir/DependInfo.cmake --color=
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f CMakeFiles/osm2pgrouting.dir/build.make CMakeFiles/osm2pgrouting.dir/build
[  3%] Linking CXX executable osm2pgrouting
/usr/local/Cellar/cmake/3.11.1/bin/cmake -E cmake_link_script CMakeFiles/osm2pgrouting.dir/link.txt --verbose=1
/usr/local/Homebrew/Library/Homebrew/shims/super/clang++   -std=c++14 -D_FILE_OFFSET_BITS=64 -DNDEBUG -Wl,-search_paths_first -Wl,-headerpad_max_install_names  CMakeFiles/osm2pgrouting.dir/src/configuration/configuration.cpp.o CMakeFiles/osm2pgrouting.dir/src/configuration/tag_key.cpp.o CMakeFiles/osm2pgrouting.dir/src/configuration/tag_value.cpp.o CMakeFiles/osm2pgrouting.dir/src/database/Export2DB.cpp.o CMakeFiles/osm2pgrouting.dir/src/database/configuration_config.cpp.o CMakeFiles/osm2pgrouting.dir/src/database/osm_nodes_config.cpp.o CMakeFiles/osm2pgrouting.dir/src/database/osm_relations_config.cpp.o CMakeFiles/osm2pgrouting.dir/src/database/osm_ways_config.cpp.o CMakeFiles/osm2pgrouting.dir/src/database/pois_config.cpp.o CMakeFiles/osm2pgrouting.dir/src/database/table_management.cpp.o CMakeFiles/osm2pgrouting.dir/src/database/ways_config.cpp.o CMakeFiles/osm2pgrouting.dir/src/database/ways_vertices_pgr_config.cpp.o CMakeFiles/osm2pgrouting.dir/src/osm_elements/Node.cpp.o CMakeFiles/osm2pgrouting.dir/src/osm_elements/OSMDocument.cpp.o CMakeFiles/osm2pgrouting.dir/src/osm_elements/Relation.cpp.o CMakeFiles/osm2pgrouting.dir/src/osm_elements/Way.cpp.o CMakeFiles/osm2pgrouting.dir/src/osm_elements/osm2pgrouting.cpp.o CMakeFiles/osm2pgrouting.dir/src/osm_elements/osm_element.cpp.o CMakeFiles/osm2pgrouting.dir/src/osm_elements/osm_tag.cpp.o CMakeFiles/osm2pgrouting.dir/src/parser/ConfigurationParserCallback.cpp.o CMakeFiles/osm2pgrouting.dir/src/parser/OSMDocumentParserCallback.cpp.o CMakeFiles/osm2pgrouting.dir/src/parser/XMLParser.cpp.o CMakeFiles/osm2pgrouting.dir/src/utilities/handle_pgpass.cpp.o CMakeFiles/osm2pgrouting.dir/src/utilities/prog_options.cpp.o CMakeFiles/osm2pgrouting.dir/src/utilities/utilities.cpp.o  -o osm2pgrouting /usr/local/lib/libpqxx.dylib /usr/local/opt/libpq/lib/libpq.dylib /usr/local/opt/expat/lib/libexpat.dylib /usr/local/lib/libboost_program_options-mt.dylib 
Undefined symbols for architecture x86_64:
  "std::__1::basic_ostream<char, std::__1::char_traits<char> >& std::__1::operator<<<char, std::__1::char_traits<char>, std::__1::allocator<char> >(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)", referenced from:
      osm2pgr::operator<<(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, osm2pgr::Tag const&) in osm_tag.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [osm2pgrouting] Error 1
make[1]: *** [CMakeFiles/osm2pgrouting.dir/all] Error 2
make: *** [all] Error 2
```

--- a/include/parser/ConfigurationParserCallback.h
+++ b/include/parser/ConfigurationParserCallback.h
@@ -23,9 +23,7 @@
 #define SRC_CONFIGURATIONPARSERCALLBACK_H_
 #pragma once
 
-#if __GNUC__ > 4 || \
-                    (__GNUC__ == 4 && (__GNUC_MINOR__ >= 6))
-#else
+#ifdef BOOST_NO_CXX11_NULLPTR
 #define nullptr NULL
 #endif
 

--- a/include/parser/OSMDocumentParserCallback.h
+++ b/include/parser/OSMDocumentParserCallback.h
@@ -23,9 +23,7 @@
 #define SRC_OSMDOCUMENTPARSERCALLBACK_H_
 #pragma once
 
-#if __GNUC__ > 4 || \
-            (__GNUC__ == 4 && (__GNUC_MINOR__ >= 6))
-#else
+#ifdef BOOST_NO_CXX11_NULLPTR
 #define nullptr NULL
 #endif
 

